examples/tests/dac/main.c

```diff
@@ -18,17 +18,20 @@ uint16_t sine_samples[100] = {
 };
 
 int main(void) {
-  int ret;
+  syscall_return_t ret;
 
   printf("[DAC] Sine test app\n");
 
   ret = dac_initialize();
-  if (ret != 0) printf("ERROR initializing DAC\n");
+  if (ret.type < TOCK_SYSCALL_SUCCESS) {
```

Comment:
I think you'd rather match on the precise variant here. _Less than_ comparison will probably work given the enum values, but that seems more like a coincidence in how the values are chosen. Also, each command is only supposed to have one failure and success variant, which would make matching on the exact value more expressive.

Reply:
Fixed